core/common/buffer.cpp

```diff
@@ -0,0 +1,107 @@
+#include <utility>
+
+#include "buffer.hpp"
+#include "hexutil.hpp"
+
+namespace kagome::common {
+
+size_t Buffer::size() const {
+  return _data.size();
+}
+
+Buffer &Buffer::put_uint32(uint32_t n) {
+  _data.reserve(sizeof(n));
+
+  _data.push_back(static_cast<unsigned char &&>((n >> 24) & 0xFF));
+  _data.push_back(static_cast<unsigned char &&>((n >> 16) & 0xFF));
+  _data.push_back(static_cast<unsigned char &&>((n >> 8) & 0xFF));
+  _data.push_back(static_cast<unsigned char &&>((n) & 0xFF));
+
+  return *this;
+}
+
+Buffer &Buffer::put_uint64(uint64_t n) {
+  _data.reserve(sizeof(n));
+
+  _data.push_back(static_cast<unsigned char &&>((n >> 56) & 0xFF));
+  _data.push_back(static_cast<unsigned char &&>((n >> 48) & 0xFF));
+  _data.push_back(static_cast<unsigned char &&>((n >> 40) & 0xFF));
+  _data.push_back(static_cast<unsigned char &&>((n >> 32) & 0xFF));
+  _data.push_back(static_cast<unsigned char &&>((n >> 24) & 0xFF));
+  _data.push_back(static_cast<unsigned char &&>((n >> 16) & 0xFF));
+  _data.push_back(static_cast<unsigned char &&>((n >> 8) & 0xFF));
+  _data.push_back(static_cast<unsigned char &&>((n) & 0xFF));
+
+  return *this;
+}
+
+const uint8_t *Buffer::to_bytes() {
+  return _data.data();
+}
+
+const std::string Buffer::to_hex() {
```

Comment:
Is there any sense in returning a `const` string instead of just string?

Reply:
yep, forgot to add const modifier to method